src/project/internal/notationproject.cpp

```diff
@@ -338,40 +340,60 @@ mu::Ret NotationProject::saveScore(const io::path& path, SaveMode saveMode)
 
 mu::Ret NotationProject::doSave(bool generateBackup)
 {
-    // Step 1: create backup if need
-    if (generateBackup) {
-        makeCurrentFileAsBackup();
+    QString currentPath = m_engravingProject->path();
+    QString savePath = currentPath + "_saving";
+
+    // Step 1: check writable
+    {
+        QFileInfo fi(savePath);
+        if (fi.exists() && !QFileInfo(savePath).isWritable()) {
+            LOGE() << "failed save, not writable path: " << savePath;
+            return make_ret(notation::Err::UnknownError);
+        }
     }
 
-    // Step 2: check writable
-    QFileInfo info(m_engravingProject->path());
-    if (info.exists() && !info.isWritable()) {
-        LOGE() << "failed save, not writable path: " << info.filePath();
-        return make_ret(notation::Err::UnknownError);
-    }
+    // Step 2: write project
+    {
+        std::string suffix = io::suffix(currentPath);
+        MscWriter::Params params;
+        params.filePath = savePath;
+        params.mode = mcsIoModeBySuffix(suffix);
+        IF_ASSERT_FAILED(params.mode != MscIoMode::Unknown) {
+            return make_ret(Ret::Code::InternalError);
+        }
 
-    // Step 3: write project
-    std::string suffix = io::suffix(info.fileName());
-    MscWriter::Params params;
-    params.filePath = m_engravingProject->path();
-    params.mode = mcsIoModeBySuffix(suffix);
-    IF_ASSERT_FAILED(params.mode != MscIoMode::Unknown) {
-        return make_ret(Ret::Code::InternalError);
+        MscWriter msczWriter(params);
+        Ret ret = writeProject(msczWriter, false);
+        if (!ret) {
+            LOGE() << "failed write project to buffer";
+            return ret;
+        }
+
+        msczWriter.close();
     }
 
-    MscWriter msczWriter(params);
-    Ret ret = writeProject(msczWriter, false);
-    if (!ret) {
-        LOGE() << "failed write project to buffer";
-        return ret;
+    // Step 3: create backup if need
+    {
+        if (generateBackup) {
+            makeCurrentFileAsBackup();
```

Comment:
But the logic in `makeCurrentFileAsBackup` is still wrong, it doesn't appear to do anything?  Line 192 (effectively an early return if the current project file *does* exist) is surely not right...(but for me even the first test in that function `if (!created().val)` was succeeding, hence returning at that point before doing anything). @igorkorsukov

Reply:
I renamed and reopened your issue